docs/cxx-proposal/cxx.rst

```diff
@@ -0,0 +1,469 @@
+C++ Proposal Details
+====================
+
+Caveats
++++++++
+
+Both the code examples and the sample header files are written without regard
+to either namespacing or how the `#include` structure will be implemented.  For
+clarity, inline code definitions are omitted from header files though in many cases
+they will be present in the final product.
+
+TL;DR
++++++
+* `Opentime C++ simplified header files <https://github.com/davidbaraff/OpenTimelineIO/tree/master/proposed-c%2B%2B-api/opentime>`_
+* `OTIO C++ simplified hear files <https://github.com/davidbaraff/OpenTimelineIO/tree/master/proposed-c%2B%2B-api/otio>`_.
+
+Dependencies
+++++++++++++
+
+The  C++ OpentimelineIO (OTIO) library implementation will have the following dependencies:
+
+    * `rapidjson <https://github.com/Tencent/rapidjson>`_
+    * any (C++ class)
+    * optional (C++ class)
+    * The C++ Opentime library (see below)
+
+The need for an "optional" (i.e. a container class that holds either no value
+or some specific value, for a given type T) is currently small, but does occur
+in one key place (schemas which need to hold either a ``TimeRange`` or
+indicate their time range is undefined).
```

Comment:
A Clip's `media_reference` field can also be null. If this is a problem, we could perhaps enforce the use of `MissingReference` instead of null.

Reply:
or perhaps I misunderstood. Is "optional" only needed for non-SerializableObject types?